### PR TITLE
Add deadline to training page

### PR DESCRIPTION
**Pull request by GuillermoFidalgo**
This one does **not** produce weird errors unlike last time. :)

### CODE REVIEW

File: _workinggroups/training.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ The [HSF-Training GitHub Organization](https://github.com/hsf-training) has [Ana
 {% for post in schools %}
   {% capture date %}{{post.end_date | date: '%s' | plus: 0 }}{% endcapture %}
   {% if date > now %}
-  1. [**{{post.date | date: "%-d %b"}} - {{post.end_date | date: "%-d %b %Y"}}** - {{post.title}} ]({{post.source}})
+  1. [**{{post.date | date: "%-d %b"}} - {{post.end_date | date: "%-d %b %Y"}}** - {{post.title}} - **Deadline:** {{post.deadline}} ]({{post.source}})
```
Comment (klieret) [inline _workinggroups/training.md]: Please add an if statement that it only shows the ``Deadline:``, if a deadline has been configured

Comment (GuillermoFidalgo) [reply _workinggroups/training.md]: Regarding the `if statement`
I have implemented the change but it produces an error that I think means that the object `post.deadline` is  an "nil" or false
```
{% for post in schools %}
  {% capture date %}{{post.end_date | date: '%s' | plus: 0 }}{% endcapture %}
  {% if date > now %}
  {%if {{post.deadline}} %}
  1. [**{{post.date | date: "%-d %b"}} - {{post.end_date | date: "%-d %b %Y"}}** - {{post.title}} - **Deadline:** {{post.deadline}} ]({{post.source}})
  {% else %}
  1. [**{{post.date | date: "%-d %b"}} - {{post.end_date | date: "%-d %b %Y"}}** - {{post.title}} ]({{post.source}})
  {% endif %}
  {% endif %}
{% endfor %}
```

image

Comment (klieret) [reply _workinggroups/training.md]: hi, this seems to be a jekyll bug as seen here: https://github.com/jekyll/jekyll/issues/7380

Adding ``excerpt: none`` to the frontmatter solves this for me.

Comment (GuillermoFidalgo) [reply _workinggroups/training.md]: Thanks so much. Now I believe we need to either remove the `deadine: ''` attribute altogether or just leave it as `deadline: ` in order to get the correct "Truthy" or "Falsy" that jekyll is looking for.

Comment (GuillermoFidalgo) [reply _workinggroups/training.md]: I have tried both of these solutions and also commenting on the line where `deadline: ''` is found and all of them work great. I will proceed to just comment them out.

**Comment by GuillermoFidalgo** [comment]
Check it out for yourself. I believe all is done.

File: _workinggroups/training.md (hunk 1 above)
Comment (klieret) [reply _workinggroups/training.md]: Hmm, I think it's not elegant to comment out the deadline field. The if statement should be `` {% if post.deadline != blank %}`` then it works.

File: _includes/list_of_schools.md [hunk 2]
```diff
@@ -2,20 +2,24 @@
...
+{% capture now %}{{'now' | date: '%s' }}{% endcapture %}
 
 ## Current and Upcoming Training Schools
 #### **Warning** : Application deadlines are **before the date shown**
 {% for post in schools %}
-  {% capture date %}{{post.end_date | date: '%s' | plus: 0 }}{% endcapture %}
+  {% capture date %}{{post.end_date | date: '%s' }}{% endcapture %}
   {% if date > now %}
+  {% if post.deadline %}
+  1. [**{{post.date | date: "%-d %b"}} - {{post.end_date | date: "%-d %b %Y"}}** - {{post.title}} - **Deadline :** {{post.deadline}} ]({{post.source}})
```
Comment (klieret) [inline _includes/list_of_schools.md]: there's a space before the colon of deadline

**Comment by klieret** [comment]
Looking great now, thanks a lot (also nice that you converted the date format)!  :)

**Comment by GuillermoFidalgo** [comment]
Ok I believe we are ready. Thanks for the  `!= blank` suggestion I did not know it was something built in.

File: _workinggroups/training.md (hunk 1 above)
Comment (GuillermoFidalgo) [reply _workinggroups/training.md]: Done!

**Pull request merged**